wandb/sdk/artifacts/artifact_manifest_entry.py

```diff
@@ -132,9 +135,13 @@ def download(self, root: Optional[str] = None) -> FilePathStr:
             cache_path = self._parent_artifact.manifest.storage_policy.load_file(
                 self._parent_artifact, self
             )
-        return FilePathStr(
-            str(filesystem.copy_or_overwrite_changed(cache_path, dest_path))
-        )
+
+        if self.skip_cache_copy:
```

Comment:
Seems like here, we're only skipping copying the downloaded files from the cache directory. We're not downloading the files to the 'dest_path'?

I think we should download the files in the `dest_path` since that's something the user can also specify (`download_root` param)

Reply:
We actually do download to the destination path. We skip the cache path download and copy entirely. The variable names are confusing. I’m passing in destination path as cache path

Reply:
Ah i see, you've used the dest path name in the s3_handler. We might need to design a better path to do this so that all handlers can support skipping the cache.

Reply:
Instead of updating each handler, you could potentially add 
```
self._parent_artifact.manifest.storage_policy._cache = `dest_path`
```
to the download() method of `ArtifactManifestEntry` in artifact_manifest_entry.py?